datafusion/physical-plan/src/filter.rs

```diff
@@ -194,11 +194,13 @@ impl ExecutionPlan for FilterExec {
     fn statistics(&self) -> Result<Statistics> {
         let predicate = self.predicate();
 
+        let input_stats = self.input.statistics()?;
         let schema = self.schema();
         if !check_support(predicate, &schema) {
-            return Ok(Statistics::new_unknown(&schema));
+            // assume worst case, that the filter is highly selective and
+            // returns all the rows from its input
+            return Ok(input_stats.clone().into_inexact());
```

Comment:
I wonder if we can make a slightly different assumption that is a better metric, e.g. each filter returning 50% or 20% of input rows?

Reply:
We could add a configuration option to control the default selectivity. I'll take a look.

Reply:
This is an interesting idea. Since statistics will be `Inexact`, it should never result in an incorrect output, but may improve average-case complexity.

Reply:
It looks like making this configurable will be a larger change. I filed https://github.com/apache/arrow-datafusion/issues/8133 and linked to it from the comment here.

Reply:
The talk [Join Order Optimization with (almost) no Statistics](https://www.youtube.com/watch?v=aNRoR0Z3SzU) is focused on full join reordering rather than just choosing the build side of a join but talks about selectivity estimates and is very relevant to this discussion. They found that selectivity of 0.2 worked well with TPC-H.

Reply:
I pushed a change to use 0.2 as the default, and now Q18 has an improved join order as well. I updated the results in the PR description.

Reply:
Default selectivities / cost estimates work ok for TPCH queries where the data is relatively uniformly distributed. 

However, in general in my experience they tend to cause problems when the data is skewed or has correlations between the columns.

Hopefully we'll be able to keep the number of hard coded constants / assumptions low in DataFusion (so there are fewer things for the optimizer to get wrong :) )